mmdet/datasets/coco.py

```diff
@@ -370,8 +386,8 @@ def evaluate(self,
         result_files, tmp_dir = self.format_results(results, jsonfile_prefix)
 
         eval_results = {}
-        cocoGt = self.coco
         for metric in metrics:
+            cocoGt = copy.deepcopy(self.coco)
```

Comment:
:smile: 
Computing segm ap preliminary encodes segmentation mask to rle. There is no known (to me) way how to get back exactly the same contour representation of segmentation masks. So that running (without deepcopy) test.py with --eval segm f1 gives different result from --eval f1 segm.